lib/clAdapters/google/oauthCalendar/index.ts

```diff
@@ -0,0 +1,292 @@
+import * as googleapis from 'googleapis';
+import * as moment from 'moment';
+import * as _ from 'lodash';
+import { Configuration, Service } from '../../base/index';
+import GoogleBaseAdapter from '../base/Adapter';
+import { GoogleError, GoogleErrorType, createGoogleError, InvalidGrant } from '../errors';
+
+const calendar = googleapis.calendar('v3');
+
+function handleGoogleError(res: Function, rej: Function, returnVal?: any) {
+  return (err: any, result: any) => {
+    if (err) {
+      let mapped = err;
+      if (err instanceof Error) {
+        // Map to custom errors
+        if (/unauthorized_client/.test(err.message.toString())) {
+          mapped = createGoogleError(
+            'UnauthorizedClient',
+            err
+          );
+        }
+        // TODO: other types
+      } else if (!err.kind) {
+        // Not a GoogleError
+        mapped = new Error(JSON.stringify(err));
+      }
+      // Leave GoogleErrors
+      rej(mapped);
+    } else {
+      res(typeof returnVal !== 'undefined' ?  returnVal : result);
+    }
+  };
+}
+
+
+
+export const fieldNameMap = {
```

Comment:
Yea I just copied them from the service account adapter; I can try putting them in the base adapter unless @cynick has a preference for how to proceed.